theano/sandbox/cuda/dnn.py

```diff
@@ -2186,6 +2186,460 @@ def method(self):
 #endif
         """
 
+class GpuDnnBatchNormBase(DnnBase):
+    """
+    Op for the cuDNN Batch Normalization.
+
+    Parameters
+    ----------
+    tensor_format
+        Always set this to 'bc01'.
+    mode
+        Normalization is performed per-activation for 'per-activation' (in this mode bnBias and bnScale tensor
+        dimensions are 1xCxHxW for 4d tensor) or Normalization is performed over N+spatial dimensions for 'spatial'
+        (in this mode bnBias, bnScale tensor dimensions are 1xCx1x1).
+    exp_avg_factor
+        Factor used in the moving average computation runningMean = newMean*factor + runningMean*(1-factor). It is
+        required that exponentialAverageFactor=1 is used for the very first call of a complete training cycle.
```

Comment:
This description indicate to me that the value should change between epoch. As we don't want to recompile Theano function between epoch to change this value, I think it should be an input of the node (in the make_node).